lib/open_food_network/feature_toggle.rb

```diff
@@ -1,21 +1,78 @@
 module OpenFoodNetwork
+  # This feature toggles implementation provides two mechanisms to conditionally enable features.
+  #
+  # You can provide an ENV var with the prefix `OFN_FEATURE_` and query it using the
+  # `ApplicationHelper#feature?` helper method. For instance, providing the ENV var
+  # `OFN_FEATURE_NEW_SHINNY_FEATURE` you could then query it from view as follows:
+  #
+  #   - if feature? :new_shinny_feature
+  #     = render "new_shinny_feature"
+  #
+  # Alternatively, you can choose which users have the feature toggled on. To do that you need to
+  # register the feature and its users from an initializer like:
+  #
+  #   require 'open_food_network/feature_toggle'
+  #   OpenFoodNetwork::FeatureToggle.enable(:new_shinny_feature, ['ofn@example.com'])
+  #
+  # It's although advisable to read the emails list from an ENV Var.
+  #
+  # You can then check it from a view like:
+  #
+  #   - if feature? :new_shinny_feature, spree_current_user
+  #     = render "new_shinny_feature"
+  #
   class FeatureToggle
-    def self.enabled?(feature_name)
-      true?(env_variable_value(feature_name))
+    def self.enabled?(feature_name, user = nil)
+      new.enabled?(feature_name, user)
+    end
+
+    def self.enable(feature_name, user_emails)
+      Thread.current[:features] ||= {}
```

Comment:
I think this Thread.current[:features] is a FeaturesStore and would be better extracted as a class.

Reply:
yes, I actually typed a class `Store` but it felt not necessary yet but I can do that and see if we can improve it a bit.

Reply:
I'd probably wait after using it for the first time. I add extra complexity that's not worth yet IMO. However, I do think we'll end up extracting that class sooner or later.